core/app/models/spree/payment_method.rb

```diff
@@ -11,6 +11,8 @@ module Spree
   # This class is not meant to be instantiated. Please create instances of concrete payment methods.
   #
   class PaymentMethod < Spree::Base
+    class UnsupportedPaymentMethod < StandardError; end
```

Comment:
I think this could be a good spot for https://api.rubyonrails.org/classes/ActiveSupport/ActionableError/ClassMethods.html, which is the same class used to run migrations directly in the browser when needed. What do you think?

Reply:
It is a good idea, I actually didn't think about it at first (and didn't know about it until now). 

But that's a Rails 6 feature and Solidus minimum requirement is for Rails 5.0.0, so it can't be implemented, am I right?

Reply:
Yes, you are right, but if we like the idea we could conditionally include `ActiveSupport::ActionableError` and its related code only when that class is defined. Up to you!

Reply:
Just to be sure, you mean like a module to include that could include all the stuff related to this PR? Including a static method to do the actual updates, the `ActiveSupport::ActionableError` and maybe the overriding of `find_sti_class` to catch the actual error?

Reply:
I mean something like:

```ruby
class UnsupportedPaymentMethod < StandardError
  if defined?(ActiveSupport::ActionableError)
    include ActiveSupport::ActionableError

    action "Bla bla" do
      Our::Rake::Task.execute
    end
  end
end

Reply:
I tried to implement it without success: the button doesn't appear if the error was raised in `ActiveRecord.find_sti_class`. I could only find base examples of this feature usage and nothing in official docs that helped my solve my specific case. I would leave this for another PR for now (and maybe I'd open an issue after a bit more digging).